ledger/src/blockstore.rs

```diff
@@ -2844,11 +2848,21 @@ impl Blockstore {
             .into_par_iter()
             .for_each(|slot| {
                 if let Ok(entries) = self.get_slot_entries(slot, 0) {
-                    entries.par_iter().for_each(|entry| {
-                        entry.transactions.iter().for_each(|tx| {
-                            tx.message.static_account_keys().iter().for_each(|pubkey| {
-                                result.insert(*pubkey);
-                            });
+                    entries.into_par_iter().for_each(|entry| {
+                        entry.transactions.into_iter().for_each(|tx| {
+                            if let Some(lookups) = tx.message.address_table_lookups() {
+                                lookups.iter().for_each(|lookup| {
+                                    result.insert(lookup.account_key);
+                                });
+                            }
+                            let sanitized_tx = bank.fully_verify_transaction(tx).unwrap();
```

Comment:
Definitely better than it was, as when I wrote this originally I didn't even know lookup tables were a thing - but I'm not 100% convinced this is correct either. If a lookup table is extended in the slot range, then a tx after that could fail to verify this check (bank is looking at beginning of range) and we wouldn't add the associated keys for the transaction.

Reply:
@apfitzge hehe, i know there's buried compromise here: how about this helpful, cheerful, encouraging comment?: db26aed92b7a101a3fb010f86b4f669860d9b8ba

Reply:
patches are welcome but it'll take some good deal of efforts to support this fully....

Reply:
^ agreed. I'm not sure how many people actually use this subcommand - but not sure its worth distracting us over scheduler.

I'd update the comment to include this issue I just made: #30165 😄

Reply:
> patches are welcome but it'll take some good deal of efforts to support this fully....

Yeah, it might be that we just have to replay the slot range and add accounts as we see them that way which is a bit of effort.
I hadn't taken that approach initially because we need to keep account state at the beginning of the snapshot range in order to create it, and process of (unpacking + replay + delete + unpacking + minimize + snapshot) seemed too long.
And also the fact ALTs weren't in (to my knowledge) at the time I started on the project as a total Solana noob haha.

With @xiangzhu70's change for keeping and loading from snapshot files will allow us to do this in a timely manner!

Reply:
> I'd update the comment to include this issue I just made: https://github.com/solana-labs/solana/issues/30165 :smile:

thanks for creating issue; just done: https://github.com/solana-labs/solana/pull/30158/commits/a2b85652f30640e56ec02ca8f20a667085d3fe7e

> With @xiangzhu70's change for keeping and loading from snapshot files will allow us to do this in a timely manner!

Oh sounds promising. :)